libs/design-system/src/atoms/inputs/toggleInputs.tsx

```diff
@@ -25,28 +22,42 @@ export type ToggleInputProps = {
   className?: string;
 
   /**
-   * The function that handles the change of the input.
+   * The function that handles a boolean and returns nothing.
    */
-  handleChange: React.ChangeEventHandler<HTMLInputElement>;
+  handleChange: (isChecked: boolean) => void;
 };
 
-export const ToggleInput = ({ options, checked = false, inputId, className, handleChange }: ToggleInputProps) => {
+export const ToggleInput = ({ options, inputId, className, handleChange }: ToggleInputProps) => {
+  const [checkedState, setCheckedState] = useState<boolean>(false);
+
+  const clickHandler = (isChecked: boolean) => {
+    setCheckedState(isChecked);
+    handleChange(isChecked);
+  };
+
   return (
-    <div className="flex space-x-4 w-36">
-      <input
+    <div
```

Comment:
Same here on not using native inputs - this could be a checkbox

Reply:
For this one, I did not even try to format the native checkbox HTML Input. I'd already lost tons of time trying to reshape the dropdown select, so, for this one, I did not bother at all.

My suggestion's the same as before: we go for it and see later if anything breaks.

What do you think?